### ISSUE TITLE
Have to manually enable/disable video streams in order to view them

### ISSUE THREAD

**Issue by benfrancis** [mozilla-iot/onvif-adapter#22]
I'm having problems viewing the video stream for a Foscam R2 camera (worked previously). I can still view still image snapshots.

In the browser console I see:
Error loading video: 
```
Object { severity: 2, category: 4, code: 4001, data: (1) […], handled: false }
video.js:140:16
    _expandVideo video.js:140

XML Parsing Error: syntax error
Location: https://foo.mozilla-iot.org/things/onvif-00623E925C8F
Line Number 1, Column 1: onvif-00623E925C8F:1:1

```

(domain and id obfuscated)

**Comment by mrstegeman** [comment]
@benfrancis Have you enabled the stream? There's a button you have to press to do that now. After that, you'll need to wait probably 20 seconds before the stream is viewable.

**Comment by benfrancis** [comment]
Oh, no I hadn't. What's the reason for the checkbox?

**Comment by mrstegeman** [comment]
I added that so that we don't have N transcode processes running at all times in the background when it's unnecessary.

Are things working after you check the box?

**Comment by benfrancis** [comment]
Ah, I see. Yes, it works after I check the box.

So, from a UX point of view I'd suggest this checkbox definitely shouldn't be needed. Clicking on the video stream in the UI should start the transcoding process. What would we need to change to make that possible?

**Comment by mrstegeman** [comment]
I agree that it's far from ideal, but it seemed better than having multiple transcodes running for no reason, slowing things down and eating power.

In theory, it's now possible for the adapter to handle the streams on its own. However, it'd involve encoding each segment as base-64, and transporting those all over IPC, which would make the already-slow experience even slower.

To make things really seamless, the gateway would probably need to:
1. Know which media files belong to which add-on
2. Be able to reliably determine when a user is (trying to) view a stream
3. Tell the add-on when its stream is being accessed
